swaylock/main.c

```diff
@@ -621,78 +614,212 @@ int main(int argc, char **argv) {
+			state->args.colors.inside.input = parse_color(optarg);
 			break;
 		case LO_INSIDE_CLEAR_COLOR:
-			state.args.colors.inside.cleared = parse_color(optarg);
+			state->args.colors.inside.cleared = parse_color(optarg);
 			break;
 		case LO_INSIDE_VER_COLOR:
-			state.args.colors.inside.verifying = parse_color(optarg);
+			state->args.colors.inside.verifying = parse_color(optarg);
 			break;
 		case LO_INSIDE_WRONG_COLOR:
-			state.args.colors.inside.wrong = parse_color(optarg);
+			state->args.colors.inside.wrong = parse_color(optarg);
 			break;
 		case LO_KEY_HL_COLOR:
-			state.args.colors.key_highlight = parse_color(optarg);
+			state->args.colors.key_highlight = parse_color(optarg);
 			break;
 		case LO_LINE_COLOR:
-			state.args.colors.line.input = parse_color(optarg);
+			state->args.colors.line.input = parse_color(optarg);
 			break;
 		case LO_LINE_CLEAR_COLOR:
-			state.args.colors.line.cleared = parse_color(optarg);
+			state->args.colors.line.cleared = parse_color(optarg);
 			break;
 		case LO_LINE_VER_COLOR:
-			state.args.colors.line.verifying = parse_color(optarg);
+			state->args.colors.line.verifying = parse_color(optarg);
 			break;
 		case LO_LINE_WRONG_COLOR:
-			state.args.colors.line.wrong = parse_color(optarg);
+			state->args.colors.line.wrong = parse_color(optarg);
 			break;
 		case LO_RING_COLOR:
-			state.args.colors.ring.input = parse_color(optarg);
+			state->args.colors.ring.input = parse_color(optarg);
 			break;
 		case LO_RING_CLEAR_COLOR:
-			state.args.colors.ring.cleared = parse_color(optarg);
+			state->args.colors.ring.cleared = parse_color(optarg);
 			break;
 		case LO_RING_VER_COLOR:
-			state.args.colors.ring.verifying = parse_color(optarg);
+			state->args.colors.ring.verifying = parse_color(optarg);
 			break;
 		case LO_RING_WRONG_COLOR:
-			state.args.colors.ring.wrong = parse_color(optarg);
+			state->args.colors.ring.wrong = parse_color(optarg);
 			break;
 		case LO_SEP_COLOR:
-			state.args.colors.separator = parse_color(optarg);
+			state->args.colors.separator = parse_color(optarg);
 			break;
 		case LO_TEXT_COLOR:
-			state.args.colors.text.input = parse_color(optarg);
+			state->args.colors.text.input = parse_color(optarg);
 			break;
 		case LO_TEXT_CLEAR_COLOR:
-			state.args.colors.text.cleared = parse_color(optarg);
+			state->args.colors.text.cleared = parse_color(optarg);
 			break;
 		case LO_TEXT_VER_COLOR:
-			state.args.colors.text.verifying = parse_color(optarg);
+			state->args.colors.text.verifying = parse_color(optarg);
 			break;
 		case LO_TEXT_WRONG_COLOR:
-			state.args.colors.text.wrong = parse_color(optarg);
+			state->args.colors.text.wrong = parse_color(optarg);
 			break;
 		default:
 			fprintf(stderr, "%s", usage);
 			return 1;
 		}
 	}
 
+	return 0;
+}
+
+static bool file_exists(const char *path) {
+	return path && access(path, R_OK) != -1;
+}
+
+static char *get_config_path(void) {
+	static const char *config_paths[] = {
+		"$HOME/.swaylock/config",
```

Comment:
Hmm, not a fan of this one

Reply:
The path? I can remove it. I just followed the same path formats that sway uses.

Reply:
On IRC, we agreed to keep the path for now to maintain consistency with other tools.